crates/bytesbuf/src/span_builder.rs

```diff
@@ -185,6 +180,37 @@ impl SpanBuilder {
         Some(span)
     }
 
+    /// Creates a span over the filled data without consuming it from the builder.
+    /// The span shares the BlockRef with the span builder, keeping the memory alive.
+    ///
+    /// Returns `None` if there is no filled data in the builder.
+    ///
+    /// # Performance
+    ///
+    /// This operation creates a new span that shares the same memory block. The operation is
+    /// relatively cheap (just cloning a BlockRef and creating a span struct) but should not
+    /// be called in tight loops if avoidable.
+    pub(crate) fn peek_filled(&self) -> Option<Span> {
```

Comment:
Peeking unfilled memory capacity is not very meaningful as a concept, so this might just be called peek() to match the method on BytesBuf. Only the actual filled part containing the data is going to ever be peeked at.